lib/matplotlib/__init__.py

```diff
@@ -824,8 +824,9 @@ def __getitem__(self, key):
     def __repr__(self):
         class_name = self.__class__.__name__
         indent = len(class_name) + 1
-        repr_split = pprint.pformat(dict(self), indent=1,
-                                    width=80 - indent).split('\n')
+        with cbook._suppress_matplotlib_deprecation_warning():
+            repr_split = pprint.pformat(dict(self), indent=1,
+                                        width=80 - indent).split('\n')
         repr_indented = ('\n' + ' ' * indent).join(repr_split)
```

Comment:
Not exactly related to this PR, but I don't understand why we don't pass `indent=indent` and skip doing the indent ourselves here.

Reply:
Perhaps(?) because the behavior of pprint's indent is less than optimal and would result in
```
RcParams({        '_internal.classic_mode': False,
         'agg.path.chunksize': 0,
         'animation.avconv_args': [],
         'animation.avconv_path': 'avconv',
...
```
see also the example at https://docs.python.org/3/library/pprint.html#pprint.PrettyPrinter with indent=4 which clearly shows this (curious?) behavior.